### PR TITLE
[pkg/stanza] improve performance

### PR DESCRIPTION
**Pull request by newly12**
**Description:** <Describe what has changed.>

The idea is to avoid making `[]byte` every time to improve performance. In fact we had previously done benchmark against v0.52.0 and recently v0.60.0 and noticed the performance regression, give it one CPU and 40 files to be ingested, emitted logs/s reduced from ~254k/s to ~140k/s. most of cpu time is used by decoding and underlying `runtime.makeslice`, the new implementation is similar to v0.52.0.

**Link to tracking Issue:** <Issue number if applicable>
#16028 

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

```
$ benchstat old.txt new.txt
name                        old time/op  new time/op  delta
FileInput/Single-10         1.30µs ±13%  1.05µs ±10%  -19.45%  (p=0.008 n=5+5)
FileInput/Glob-10           6.32µs ±12%  4.25µs ± 8%  -32.73%  (p=0.008 n=5+5)
FileInput/MultiGlob-10      5.98µs ± 3%  4.17µs ± 9%  -30.29%  (p=0.008 n=5+5)
FileInput/MaxConcurrent-10  5.42µs ± 4%  4.26µs ±15%  -21.54%  (p=0.008 n=5+5)
FileInput/FngrPrntLarge-10  1.21µs ± 1%  1.06µs ±12%  -12.52%  (p=0.008 n=5+5)
FileInput/FngrPrntSmall-10  1.34µs ± 7%  1.00µs ± 6%  -25.32%  (p=0.008 n=5+5)
```

image

### CODE REVIEW

File: pkg/stanza/operator/helper/encoding.go [hunk 1]
```diff
@@ -19,10 +19,11 @@ import (
 	"fmt"
 	"strings"
 
+	"golang.org/x/text/transform"
+
 	"golang.org/x/text/encoding"
 	"golang.org/x/text/encoding/ianaindex"
 	"golang.org/x/text/encoding/unicode"
-	"golang.org/x/text/transform"
```
Comment (djaglowski) [inline pkg/stanza/operator/helper/encoding.go:25]: This rearrangement of the imports will have to be reverted.

**Review by djaglowski — COMMENTED** [review]
Great find, @newly12. Thanks for the improvement.

File: pkg/stanza/operator/helper/encoding.go (hunk 1 above)
Comment (newly12) [reply pkg/stanza/operator/helper/encoding.go]: fixed.

**Review by djaglowski — APPROVED** [review]
Thank you @newly12

**Pull request merged**